data_model/src/domain.rs

```diff
@@ -82,39 +145,95 @@ impl Ord for Domain {
 }
 
 impl Domain {
-    /// Construct [`Domain`] from [`Id`].
-    pub fn new(id: Id) -> Self {
-        Self {
-            id,
-            accounts: AccountsMap::new(),
-            asset_definitions: AssetDefinitionsMap::new(),
-            metadata: Metadata::new(),
-            logo: None,
-        }
+    /// Construct new empty [`Domain`] from [`Id`].
+    pub fn new(id: Id) -> NewDomain {
+        NewDomain::new(id)
     }
 
-    /// Domain constructor with pre-setup accounts. Useful for testing purposes.
-    pub fn with_accounts(name: &str, accounts: impl IntoIterator<Item = Account>) -> Self {
-        let accounts_map = accounts
-            .into_iter()
-            .map(|account| (account.id.clone(), account))
-            .collect();
-        #[allow(clippy::expect_used)]
-        Self {
-            id: Id::new(name).expect("must be valid"),
-            accounts: accounts_map,
-            asset_definitions: AssetDefinitionsMap::new(),
-            metadata: Metadata::new(),
-            logo: None,
-        }
+    /// Returns a reference to the account corresponding to the account id.
+    pub fn get_account(&self, account_id: &<Account as Identifiable>::Id) -> Option<&Account> {
+        self.accounts.get(account_id)
+    }
+
+    pub fn get_asset_definition(
+        &self,
+        asset_definition_id: &<AssetDefinition as Identifiable>::Id,
+    ) -> Option<&AssetDefinitionEntry> {
+        self.asset_definitions.get(asset_definition_id)
+    }
+
+    pub fn asset_definitions(&self) -> impl ExactSizeIterator<Item = &AssetDefinitionEntry> {
+        self.asset_definitions.values()
     }
 }
 
+#[cfg(feature = "mutable_api")]
```

Comment:
This won't work. You need to mark each `mut` function individually inside a single `impl` block.

Reply:
I tried it out and as far as I can see you're wrong. Why do you think it won't work?

Reply:
Yeah, you're right, it didn't use to work; I saw an error message with `multiple impl blocks` in static analysis and assumed it was because of this. I am indeed wrong. 

https://github.com/rust-lang/rust/issues/29322